### PR TITLE
Overworked library toward 3.0.0 release

### PR DESCRIPTION
**Pull request by nlohmann**
- fixed typos
- updated README
- ~~updated Catch~~ (version 2.0.1 is slower than the 1.9.x version we use)

* * *

## Pull request checklist

- [ ]  Changes are described in the pull request, or an [existing issue is referenced](https://github.com/nlohmann/json/issues).
- [ ]  The test suite [compiles and runs](https://github.com/nlohmann/json/blob/develop/README.md#execute-unit-tests) without error.
- [ ]  [Code coverage](https://coveralls.io/github/nlohmann/json) is 100%. Test cases can be added by editing the [test suite](https://github.com/nlohmann/json/tree/develop/test/src).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -805,13 +807,13 @@ THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR I
...
 
 ## Contact
 
 If you have questions regarding the library, I would like to invite you to [open an issue at Github](https://github.com/nlohmann/json/issues/new). Please describe your request, problem, or question as detailed as possible, and also mention the version of the library you are using as well as the version of your compiler and operating system. Opening an issue at Github allows other users and contributors to this library to collaborate. For instance, I have little experience with MSVC, and most issues in this regard have been solved by a growing community. If you have a look at the [closed issues](https://github.com/nlohmann/json/issues?q=is%3Aissue+is%3Aclosed), you will see that we react quite timely in most cases.
 
-Only if your request would contain confidential information, please [send me an email](mailto:mail@nlohmann.me).
+Only if your request would contain confidential information, please [send me an email](mailto:mail@nlohmann.me). For encrypted messages, please usse [this key](https://keybase.io/nlohmann/pgp_keys.asc).
```
Comment (gregmarr) [inline README.md]: @nlohmann typo: usse

**Comment by gregmarr** [comment]
I noticed that you were cleaning up spaces around the `<` and `>` in `template`s, but there are still a few left.  587, 592, 621, 885, 1000, 1614, 9909

File: README.md (hunk 1 above)
Comment (nlohmann) [reply README.md]: Thanks. I shall fix this.

**Comment by nlohmann** [comment]
@gregmarr This is due to Artistic Style which sometimes seems to interpret `<` and `>` as comparisons... :-(

**Comment by gregmarr** [comment]
Oh, it's an auto formatter?

**Comment by nlohmann** [comment]
Yes, see http://astyle.sourceforge.net. I am using it for some years now and have not found an equivalent Clang-format style. "My" style is implemented in the `make pretty` target.

**Pull request merged**